src/query/plan/physical.go

```diff
@@ -97,8 +97,8 @@ func (p PhysicalPlan) shiftTime() PhysicalPlan {
 		}
 
 		spec := boundOp.Bounds()
-		if p.LookbackDuration > maxOffset {
-			maxOffset = p.LookbackDuration
+		if spec.Offset > maxOffset {
```

Comment:
I think this would still have to check lookback duration wouldn't it? Maybe `if spec.Offset + p.LoobackDuration > maxOffset` works?

Reply:
See above.